operator/src/network/controller.rs

```diff
@@ -963,6 +973,54 @@ async fn reset_bootstrap_job(
     Ok(())
 }
 
+async fn apply_pod_monitor(
+    cx: Arc<Context<impl IpfsRpcClient, impl RngCore, impl Clock>>,
+    network: Arc<Network>,
+    monitor_name: &str,
+    monitor_port: u32,
+    monitor_label: &str,
+) -> Result<(), Error> {
+    // Create the pod monitor CR only if the CRD already exists
+    let crds: Api<CustomResourceDefinition> = Api::all(cx.k_client.clone());
+    let crd = crds.get(PodMonitor::crd_name()).await;
+    if crd.is_ok() {
+        // Instantiate the pod monitor CR if it doesn't already exist
+        let ns = "keramik-".to_owned() + &network.name_any();
+        let pod_monitors: Api<PodMonitor> = Api::namespaced(cx.k_client.clone(), ns.as_str());
+        let pod_monitor = pod_monitors.get_opt(monitor_name).await?;
+        if pod_monitor.is_none() {
+            let mut pod_monitor = PodMonitor::new(
+                monitor_name,
+                PodMonitorSpec {
+                    pod_metrics_endpoints: vec![PodMetricsEndpointSpec {
+                        interval: Some("10s".to_owned()),
+                        path: Some("/metrics".to_owned()),
+                        target_port: Some(monitor_port),
+                    }],
+                    selector: Some(SelectorSpec {
+                        match_labels: {
+                            let mut map = BTreeMap::new();
+                            map.insert("app".to_owned(), monitor_label.to_owned());
+                            map
+                        },
+                    }),
+                },
```

Comment:
We should add another section to the podmonitor,   `podTargetLabels`

If should be a list of strings for pod labels that should be propagated to the metrics as labels.

The list could come from the Network definition, or the static list I'm using:
```
  podTargetLabels:
    - affinity
    - scenario
    - simulation
  ```

Reply:
Ah yes, thanks for reminding me. I thought I was forgetting something 😊